src/Jmikola/AutoLogin/Http/Firewall/AutoLoginListener.php

```diff
@@ -113,7 +124,8 @@ public function handle(GetResponseEvent $event)
             }
         } catch (AuthenticationException $e) {
             if (null !== $this->logger) {
-                $this->logger->warn(
+                $warning = $this->logger instanceof PsrLoggerInterface ? 'warning' : 'warn';
+                $this->logger->$warning(
```

Comment:
@jmikola 

How about extracting a temporary variable for the log message and explicitly invoking the appropriate method instead of using a dynamic method name?